src/utils/RedisUtils.ts

```diff
@@ -0,0 +1,19 @@
+import { createClient, RedisClient } from 'redis';
+import { promisify } from 'util';
+import { IMap } from './Types';
+
+export const getAsync = (client: RedisClient) => promisify(client.get).bind(client);
+export const setAsync = (client: RedisClient) => promisify(client.set).bind(client);
+export const getObjectAsync = (client: RedisClient) => promisify(client.hgetall).bind(client);
+export const setObjectAsync = (client: RedisClient) => async <T>(key: string, values: IMap<T>) =>
+    new Promise((resolve, reject) => {
+        client.hmset(key, values as any, (err, reply) => {
+            if (err) return reject(err);
+            return resolve(reply);
+        });
+    });
+export const createRedisClient = () => createClient({
```

Comment:
I can see that this code is in two places - here and the server.ts. Does it need to be?
Also if the connection was to fail or timeout, what happens then?

Reply:
Oops that's a good point